mbtiles.go

```diff
@@ -114,28 +114,47 @@ func (db *MBtiles) ReadTile(z int64, x int64, y int64, data *[]byte) error {
 		return errors.New("cannot read tile from closed mbtiles database")
 	}
 
+	var err error
+	maxRetries := 3 // Maximum number of retries
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		err = db.readTileAttempt(z, x, y, data)
+		if err == nil {
+			return nil // Success, exit the loop
+		}
+		// Optionally, log the error and retry
+		fmt.Printf("Attempt %d failed: %v\n", attempt+1, err)
+		time.Sleep(time.Second * 2) // Wait for 2 seconds before retrying
+	}
+	return err // Return the last error encountered
+}
+
+// readTileAttempt tries to read a tile once
+func (db *MBtiles) readTileAttempt(z int64, x int64, y int64, data *[]byte) error {
 	con, err := db.getConnection(context.TODO())
-	defer db.closeConnection(con)
 	if err != nil {
 		return err
 	}
+	defer db.closeConnection(con)
 
-	query, err := con.Prepare("select tile_data from tiles where zoom_level = $z and tile_column = $x and tile_row = $y")
+	query, err := con.Prepare("select tile_data from tiles where zoom_level = $1 and tile_column = $2 and tile_row = $3")
```

Comment:
Why the change from z, x, y to 1, 2, 3?

Reply:
I've reverted that particular bit back to what i should have left it as